lib/client.js

```diff
@@ -143,11 +143,18 @@ class Client extends EventEmitter {
       }
     }
 
-    this[kQueue] = Q((request, cb) => {
+    const dispatch = (request, cb) => {
       if (this.destroyed) {
         return cb(new Error('The client is destroyed'))
       }
 
+      if (this.socket.destroyed) {
+        this[kQueue].unshift(request, cb)
+        this[kQueue].pause()
+        cb()
```

Comment:
The problem here is that we just want to reschedule the task without invoking the outer callback...

Reply:
Maybe something can be added to the fastq API in this regard?

Reply:
take a look and see if it’s feasible